testing/util/wait_timeout.go

```diff
@@ -20,3 +22,35 @@ func WaitTimeout(wg *sync.WaitGroup, timeout time.Duration) bool {
 		return true
 	}
 }
+
+type Waiter struct {
+	c chan struct{}
+}
+
+func NewWaiter() *Waiter {
```

Comment:
Not sure if applies since part of tests, but comments on exported functions/objects for this

Reply:
:+1: added comments